ansible_collections/arista/avd/roles/eos_config_deploy_cvp/templates/cvp-devices-v3.j2

```diff
@@ -5,15 +5,17 @@
 {% else %}
 {%    set local_var.device_filter = device_filter %}
 {% endif %}
+{% set dev_cntr = namespace(value=0) %}
 ---
 CVP_DEVICES:
 {% for device in groups[container_root] | arista.avd.natural_sort %}
 {%     if device | arista.avd.is_in_filter(local_var.device_filter) %}
+{%         set dev_cntr.value = dev_cntr.value + 1 %}
```

Comment:
I think it's better to move this line of code to under below if condition
```{%         if hostvars[device]['is_deployed'] is arista.avd.defined(true) or hostvars[device]['is_deployed'] is not arista.avd.defined %}```

If the above condition fails then also CVP_DEVICES will be returned None.

Reply:
yea, that's how I did it initially, not sure why I changed my mind, but that's a good point, let me move it there, thanks!